numba/np/arraymath.py

```diff
@@ -665,6 +666,136 @@ def _trivial_isnan(x):
         return _trivial_isnan
 
 
+@register_jitable
+def _impl_iscomplex_isreal_arr(fn, x):
+    ax = np.asarray(x) # NumPy uses asanyarray here!
+    return fn(ax.imag, 0)
+
+
+@register_jitable
+def np_iscomplex_isreal(x, operator_fn, opt_ret):
+
+    if type_can_asarray(x):
+        return lambda x: _impl_iscomplex_isreal_arr(operator_fn, x)
+    elif isinstance(x, types.Optional):
+        def impl(x):
+            if x is None:
+                return opt_ret
+            return _impl_iscomplex_isreal_arr(operator_fn, x)
```

Comment:
Why is this needed?

Reply:
You are correct, I will remove this part

Reply:
Hey @eric-wieser, you're right. This logic can be moved to the jitted function. I've updated the code.